stbt-batch.d/run

```diff
@@ -1,315 +1,181 @@
+                # "Uninteresting" failures due to the test infrastructure
+                continue
+            elif args.keep_going >= 2:
+                continue
+            else:
+                stop[0] = True
+
+        if args.run_once:
+            break
+
+    if run_count == 1:
+        # If we only run a single test a single time propagate the result
+        # through
+        return last_exit_status
+    elif failure_count == 0:
+        return 0
+    else:
         return 1
 
-      # Even if the card has no video connected to its input you see
-      # "VideoInputFrameArrived: Frame received - No input signal detected"
-      elif ! grep -q VideoInputFrameArrived decklinksrc.log; then
-        echo "Blackmagic card froze" > failure-reason
-        cp failure-reason failure-reason.manual
-        echo "Blackmagic card froze; exiting."
-        return 1
-      fi
-      ;;
-  esac
-}
-
-# Input:
-#   test1.py arg1 arg2 -- test2.py arg -- test3.py
-# Output (suitable as input to `IFS=$'\t' read -a`):
-#   test1.py\targ1\targ2
-#   test2.py\targ
-#   test3.py
-parse_test_args() {
-  if [[ "$*" =~ -- ]]; then
-    [[ "$1" == -- ]] && shift
-    while [[ $# -gt 0 ]]; do
-      printf "%s" "$1"; shift
-      while [[ $# -gt 0 && "$1" != -- ]]; do printf "\t%s" "$1"; shift; done
-      printf "\n"
-      [[ "$1" == -- ]] && shift
-    done
-  else  # No "--": Each command-line argument is a test script.
-    printf "%s\n" "$@"
-  fi
-}
-
-# http://stackoverflow.com/questions/392022/best-way-to-kill-all-child-processes
-killtree() {
-  local parent=$1 child
-  for child in $(ps -o ppid= -o pid= | awk "\$1==$parent {print \$2}"); do
-    killtree $child
-  done
-  kill $parent
-}
-
-# Portable implementation of GNU "readlink -f" to support BSD/OSX.
-realpath() {
-  python -c 'import os, sys; print os.path.realpath(sys.argv[1])' "$1"
-}
 
-main "$@"
+def listsplit(l, v):
+    """
+    A bit like str.split, but for lists
+    """
+    out = []
+    sublist = []
+    for x in l:
+        if x == v:
+            if sublist:
+                out.append(sublist)
+            sublist = []
+        else:
+            sublist.append(x)
+    if sublist:
+        out.append(sublist)
+    return out
+
+
+def parse_test_args(args):
+    """
+    >>> parse_test_args(['test 1.py', 'test2.py', 'test3.py'])
+    [['test1.py'], ['test2.py'], ['test3.py']]
+    >>> parse_test_args(['test1.py', 'test2.py'])
+    [['test1.py'], ['test2.py']]
+    >>> parse_test_args(['test1.py', '--'])
+    [['test1.py']]
+    >>> parse_test_args(['test1.py', '--', 'test2.py'])
+    [['test1.py'], ['test2.py']]
+    >>> parse_test_args(['test1.py', '--', 'test2.py', '--'])
+    [['test1.py'], ['test2.py']]
+    >>> parse_test_args(['test1.py', 'test2.py'])
+    >>> parse_test_args(
+    ...     ['test1.py', 'arg1', 'arg2', '--', 'test2.py', 'arg', '--',
+    ...      'test3.py'])
+    [['test1.py', 'arg1', 'arg2'], ['test2.py', 'arg'], ['test3.py']]
```

Comment:
I suspect this doctest isn't being run:

* The output at line 160 should have a space in the filename.
* Line 161 is identical to line 169 but they have different outputs.